charts/k8ssandra/templates/traefik.ingressroutes.yaml

```diff
@@ -43,3 +45,92 @@ spec:
         - name: {{ $clusterName }}-{{ $datacenterName }}-service
           port: 9042
 {{- end }}
+{{- if $traefik.stargate.enabled }}
+---
+{{- if and $traefik.stargate.cassandra.enabled $traefik.cassandra.enabled }}
+  {{- fail "ingress.traefik.stargate.cassandra.enabled and ingress.traefik.cassandra.enabled cannot both be enabled." }}
+{{- end }}
+{{- $stargateHost := default "localhost" $traefik.stargate.host }}
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: k8ssandra-stargate
+  labels:
+{{ include "k8ssandra.labels" . | indent 4 }}
+spec:
+  rules:
+    - host: {{ $stargateHost }}
```

Comment:
No, `$stargateHost` is assigned with a default value of `localhost` on line 53. The main `values.yaml` also specifies a default.

I could be convinced that each route should have its own overridable host value, falling back to the default if unspecified.

Reply:
Looking at the Kubernetes [concept docs](https://kubernetes.io/docs/concepts/services-networking/ingress/) for ingress I see can see a number of realistic configurations.

1. Developer installation
    * Most likely a locally installed k8s
    * Very little control over DNS, maybe /etc/hosts access, probably not
 2. Multi-node / Off-host k8s cluster
     * Most likely from a cloud provider or externally manager
     * Ability to control DNS
 
With _option 1_ I feel like everything should be path-based. To that end, our default configuration for ingress (including monitoring) should do all routing based on path prefixes.

For _Option 2_ I could see a couple of configurations. If it's possible to influence DNS then that would be the preferred approach. We route to Stargate backends with hostname matching and don't need to bother with defining paths. We _could_ specify paths, but it doesn't really add much IMHO (this assumption may be wrong). Should DNS be locked down then we can still deploy a config just like _option 1_.

Reply:
I learned a couple weeks ago from @jdonenine about [nip.io](https://nip.io/). It's a free, publicly available DNS system that reads the IP out of the hostname dynamically -- so you can do something like stargate.127.0.0.1.nip.io and it will resolve to 127.0.0.1. Extremely useful for devs who don't have /etc/hosts access (and even for those who do)!

Reply:
After spending some time reviewing and letting things sink in, I like the idea of providing options to address both scenarios. 

For a local, dev setup, e.g., kind, path-based routing seems like a good default. It frees me from have to worry about hostnames and more complex ingress configurations.

For the second scenario where the user may have ability to control DNS, I like allowing the user to specify different hostnames for each service and using host+path routing. The user could choose to use the same or different hostnames across services.